packages/web/src/pages/sign-up-page/components/AccountHeader.tsx

```diff
@@ -1,80 +1,173 @@
-import { Avatar, Box, Flex, Text } from '@audius/harmony'
+import { Avatar, Box, Flex, IconCamera, IconImage, Text } from '@audius/harmony'
 
 import {
   getCoverPhotoField,
-  getHandleField,
   getNameField,
   getProfileImageField
 } from 'common/store/pages/signon/selectors'
 import { useMedia } from 'hooks/useMedia'
 import { useSelector } from 'utils/reducer'
 
-export const AccountHeader = () => {
+import { ImageField } from './ImageField'
+
+type AccountHeaderProps = {
+  mode: 'editing' | 'viewing'
+}
+
+const CoverPhotoBox = ({
+  imageUrl,
+  isEditing
+}: {
+  imageUrl: string | undefined
+  isEditing?: boolean
+}) => (
+  <Box
+    h='100%'
+    w='100%'
+    border='default'
+    css={{
+      backgroundColor: 'lightgray',
+      overflow: 'hidden',
+      ...(imageUrl
+        ? {
+            backgroundImage: `url(${imageUrl})`,
+            backgroundPosition: 'center',
+            backgroundSize: '100%',
+            backgroundRepeat: 'no-repeat, no-repeat'
+          }
+        : {
+            background:
+              'linear-gradient(0deg, rgba(0, 0, 0, 0.20) 0%, rgba(0, 0, 0, 0.00) 100%), #C2C0CC'
+          })
+    }}
+  >
+    {isEditing && !imageUrl ? (
+      <IconImage
+        css={{ position: 'absolute', right: '16px', top: '16px' }}
+        color='staticWhite'
+      />
+    ) : null}
+  </Box>
+)
+
+const ProfileImageAvatar = ({
+  imageUrl,
+  isEditing
+}: {
+  imageUrl: string
+  isEditing?: boolean
+}) => {
+  const { isMobile } = useMedia()
+  const useSmallSize = isEditing || isMobile
+
+  const avatarSize = useSmallSize ? 72 : 120
+  return (
+    <Avatar
+      variant='strong'
+      src={imageUrl}
+      css={{
+        height: avatarSize,
+        width: avatarSize,
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        ...(useSmallSize ? { transform: 'translateY(20px)' } : null)
+      }}
+    >
+      {isEditing && !imageUrl ? (
+        <IconCamera size='l' color='staticWhite' />
+      ) : null}
+    </Avatar>
+  )
+}
+
+export const AccountHeader = ({ mode }: AccountHeaderProps) => {
   const coverPhoto = useSelector(getCoverPhotoField)
   const profileImage = useSelector(getProfileImageField)
   const { value: displayName } = useSelector(getNameField)
-  const { value: handle } = useSelector(getHandleField)
+  // const { value: handle } = useSelector(getHandleField)
 
-  const { isMobile } = useMedia()
+  const handle = 'f&$kTheFireNation'
+  const isEditing = mode === 'editing'
 
-  const avatarSize = isMobile ? 72 : 120
+  const { isMobile } = useMedia()
+  const useSmallSize = isEditing || isMobile
```

Comment:
reserve use prefix for hooks
prefix booleans with 'is' per style guide

Reply:
ah good catch